File: Python-version/organism.py
from abc import ABC, abstractmethod


class Organism:
    def __init__(self, x, y, world, oid=None, s=None, i=None, a=None, m=None):
        #te inne argumenty są do loadowania mapki
        self.xPos = x
        self.yPos = y
        self.world = world
        self.str = s
        self.ini = i
        self.age = 0
        self.oId = oid
        self.moved = False
        self.speciesId=0
        self.log=" "
        self.name=" "
        self.color=""#tu sb cos n bedzie zgadzac
        self.sign=""


    @abstractmethod
    def Win(self,othercreature):
        pass
    @abstractmethod
    def Defended(self,attackingcreature):
        pass
    @abstractmethod
    def TakeAction(self):
        pass
    @abstractmethod
    def print(self):
        pass

    def GetS(self):
        return self.str

    def SetS(self, strength):
        self.str=strength

    def GetI(self):
        return self.ini

    def GetX(self):
        return self.xPos

    def GetY(self):
        return self.yPos

    def GetSId(self):
        return self.speciesId

    def GetOId(self):
        return self.oId

    def SetOId(self, oid):
        self.oId=oid

    def GetAge(self):
        return self.age

    def SetAge(self, a):
        self.age=a

    def HasMoved(self):
        return self.moved

    def SetHasMoved(self, m):
        self.moved=m

    def GetLog(self):
        return self.log

    def ResetLog(self):
        self.log=" "

    def GetName(self):
        return self.name

File: Java-version/Organism.java
import java.awt.*;
abstract public class Organism
{
    public Organism(int x, int y, World w)
    {
        xPos = x;
        yPos = y;
        world = w;
    }

    public Organism(int s, int i, int x, int y, int a, int oid, boolean m, World w)
    {
        str = s;
        ini = i;
        xPos = x;
        yPos = y;
        age = a;
        oId = oid;
        moved = m;
        world = w;
    }

    public abstract void Win(Organism otherCreature);
    public abstract boolean Defended(Organism attackingCreature);
    public abstract void TakeAction();
    public abstract void Print();

    public int GetS() {
        return str;
    }
    public void SetS(int strength) {
        str = strength;
    }
    public int GetI() {
        return ini;
    }
    public int GetX() {
        return xPos;
    }
    public int GetY() {
        return yPos;
    }
    public int GetSId() {
        return speciesId;
    }
    public int GetOId() {
        return oId;
    }
    public void SetOId(int oid) {
        oId = oid;
    }
    public int GetAge() {
        return age;
    }
    public void SetAge(int a) {
        age = a;
    }
    public boolean HasMoved() {
        return moved;
    }
    public void SetHasMoved(boolean m) {
        moved = m;
    }
    public String GetLog() {
        return log;
    }
    public void ResetLog() {
        log = " ";
    }
    public String GetName() {
        return name;
    }
    protected int str, ini, xPos, yPos, age = 0, speciesId, oId;
    protected World world;
    protected char sign;
    protected Color color;
    protected String name;
    protected boolean moved = true;
    public final int DEAD = -1;
    protected String log = " ";
};
